openforcefield/tests/test_parameters.py

```diff
@@ -908,36 +908,91 @@ def test_multi_term_proper_torsion_bad_units(self):
+    def test_torsion_handler_charmm_potential():
+        """
+        Test creation of TorsionHandlers with the deprecated 0.2 potential value "charmm" instead of the current
+        supported potential value "fourier".
+        """
+        import re
+
+        # Test creating ProperTorsionHandlers
+        err_msg = re.escape("Attempted to set ProperTorsionHandler.potential to charmm. Currently, "
+                            "only the following values are supported: ['k*(1+cos(periodicity*theta-phase))'].")
+        with pytest.raises(SMIRNOFFSpecError, match=err_msg):
+            ph1 = ProperTorsionHandler(potential='charmm', skip_version_check=True)
+        ph1 = ProperTorsionHandler(potential='k*(1+cos(periodicity*theta-phase))', skip_version_check=True)
+
+        # Same test, but with ImproperTorsionHandler
+        err_msg = re.escape("Attempted to set ImproperTorsionHandler.potential to charmm. Currently, "
+                            "only the following values are supported: ['k*(1+cos(periodicity*theta-phase))'].")
+        with pytest.raises(SMIRNOFFSpecError, match=err_msg):
+            ph1 = ImproperTorsionHandler(potential='charmm', skip_version_check=True)
+        ph1 = ImproperTorsionHandler(potential='k*(1+cos(periodicity*theta-phase))', skip_version_check=True)
+
+
+class TestGBSAHandler:
+    def test_create_default_gbsahandler(self):
+        """Test creation of an empty GBSAHandler, with all default attributes"""
+        from simtk import unit
+        gbsa_handler = GBSAParameterHandler(skip_version_check=True)
+        assert gbsa_handler.gb_model == 'OBC1'
+        assert gbsa_handler.solvent_dielectric == 78.5
+        assert gbsa_handler.solute_dielectric == 1
+        assert gbsa_handler.sa_model == 'ACE'
+        assert gbsa_handler.surface_area_penalty == 5.4 * unit.calorie / unit.mole / unit.angstrom**2
+        assert gbsa_handler.solvent_radius == 1.4 * unit.angstrom
+
+
+    def test_gbsahandler_setters(self):
+        """Test creation of an empty GBSAHandler, with all default attributes"""
+        from simtk import unit
+        gbsa_handler = GBSAParameterHandler(skip_version_check=True)
+
+        gbsa_handler.gb_model = 'OBC2'
+        gbsa_handler.gb_model = 'HCT'
+        gbsa_handler.gb_model = 'OBC1'
+        with pytest.raises(SMIRNOFFSpecError) as excinfo:
+            gbsa_handler.gb_model = 'Something invalid'
+
+        gbsa_handler.solvent_dielectric = 50.0
+        gbsa_handler.solvent_dielectric = "50.0"
+        with pytest.raises(ValueError) as excinfo:
+            gbsa_handler.solvent_dielectric = 'string that can not be cast to float'
+
+        gbsa_handler.solute_dielectric = 2.5
+        gbsa_handler.solute_dielectric = "3.5"
+        with pytest.raises(ValueError) as excinfo:
+            gbsa_handler.solute_dielectric = 'string that can not be cast to float'
+
+        gbsa_handler.sa_model = 'ACE'
+        gbsa_handler.sa_model = None
```

Comment:
Do we ever want to allow for None? @jchodera

Reply:
Yes -- Setting the `sa_model` to `None` zeroes the surface area energy term.